great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -1690,12 +1691,18 @@ def delete_datasource(
 
         datasource = self.get_datasource(datasource_name=datasource_name)
 
-        if save_changes and not isinstance(datasource, FluentDatasource):
+        if isinstance(datasource, FluentDatasource):
+            # Note: this results in some unnecessary dict lookups
+            self._delete_fluent_datasource(datasource_name)
+        elif save_changes:
             datasource_config = datasourceConfigSchema.load(datasource.config)
             self._datasource_store.delete(datasource_config)
         self._cached_datasources.pop(datasource_name, None)
         self.config.datasources.pop(datasource_name, None)  # type: ignore[union-attr]
 
+        if save_changes:
+            self._save_project_config()
```

Comment:
This is new behavior. Previously we were not updating `great_expectations.yml` when `deleting` a datasource, even if `save_changes=True`.